controller/src/test/java/io/pravega/controller/store/client/StoreClientFactoryTest.java

```diff
@@ -70,24 +67,10 @@ public void testZkSessionExpiryRetry() throws Exception {
         
         sessionExpiry.join();
 
-        Supplier<Boolean> isAliveSupplier = () -> {
-            try {
-                return client.getZookeeperClient().getZooKeeper().getState().isAlive();
-            } catch (Exception e) {
-                throw new CompletionException(e);
-            }
-        };
-
-        Futures.loop(isAliveSupplier,
```

Comment:
we could write this same loop using assertEventuallyEquals.. but why have we removed this? do we no longer need to wait on this?

Reply:
My understanding is, with `injectSessionExpiration` the Curator actually fakes a session expiration and so checking the State of the real Zookeeper instance is not needed here and the busy wait can actually cause the test to timeout unnecessarily because the two things are internally unrelated.
Ref:
https://stackoverflow.com/questions/41862575/how-does-apache-curator-try-to-reconnect-when-session-expires
http://curator.apache.org/errors.html ( See under "Notifications" Zookeeper state=LOST)

Reply:
Code for ZookeeperTestable.injectSessionExpiration :
```
class ZooKeeperTestable implements Testable {
    private static final Logger LOG = LoggerFactory
            .getLogger(ZooKeeperTestable.class);

    private final ZooKeeper zooKeeper;
    private final ClientCnxn clientCnxn;

    ZooKeeperTestable(ZooKeeper zooKeeper, ClientCnxn clientCnxn) {
        this.zooKeeper = zooKeeper;
        this.clientCnxn = clientCnxn;
    }

    @Override
    public void injectSessionExpiration() {
        LOG.info("injectSessionExpiration() called");

        clientCnxn.eventThread.queueEvent(new WatchedEvent(
                Watcher.Event.EventType.None,
                Watcher.Event.KeeperState.Expired, null));
        clientCnxn.eventThread.queueEventOfDeath();
        clientCnxn.state = ZooKeeper.States.CLOSED;
        clientCnxn.sendThread.getClientCnxnSocket().onClosing();
    }
}
```

Reply:
Added assertEventuallyThrows() call  as suggested above to the test to make sure we attempt get the exception multiple times, subject to a timeout.